lib/gem_ext/rails/reflection.rb

```diff
@@ -0,0 +1,59 @@
+require 'active_record/reflection'
+
+ActiveRecord::Reflection.module_eval do
+  def self.create(macro, name, scope, options, ar)
```

Comment:
Yeah I wish the way associations were hooked was a little more extensible so I didn't have to do as much monkey patching to make it work.